js/leanModal.js

```diff
@@ -20,15 +19,25 @@
         dismissible: true,
         starting_top: '4%'
       },
-      overlayID = _generateID(),
       $modal = $(this),
-      $overlay = $('<div class="lean-overlay"></div>'),
       lStack = (++_stack);
-
-      // Store a reference of the overlay
-      $overlay.attr('id', overlayID).css('z-index', 1000 + lStack * 2);
-      $modal.data('overlay-id', overlayID).css('z-index', 1000 + lStack * 2 + 1);
-
+      
+      if ($modal.css("display") == "block") {
+        // The modal is already opened, just put it back on top
+        associatedOverlayID = $modal.data('overlay-id');
+        $associatedOverlay = $('#' + associatedOverlayID);
+        $associatedOverlay.css('z-index', 1000 + lStack * 2);
+        $modal.css('z-index', 1000 + lStack * 2 + 1);
+        return;
+      }
+      else {
```

Comment:
Since the previous if statement ends in a return there is no point to have an else statement here. What's encapsulated within the else curly brackets is no-way separated from the rest of the function.